apps/aefate/src/aefa_data.hrl

```diff
@@ -0,0 +1,91 @@
+%% Common definitions.
+
+-define(FATE_UNIT,       {tuple, {}}).
+-define(FATE_TUPLE(T),   {tuple, T}).
+-define(FATE_ADDRESS(A), {address, A}).
+
+-ifdef(CHECKFATEDATA).
+
+%% Types
+-define(FATE_INTEGER_T, {integer,  integer()}).
+-define(FATE_BYTE_T,    {integer,  0..255}).
+-define(FATE_BOOLEAN_T, {boolean, true | false}).
+-define(FATE_NIL_T,     {nil}).
+-define(FATE_LIST_T(T), {list,    list(A::T) | ?FATE_NiL_T}).
+-define(FATE_UNIT_T,    {tuple,   {}}).
+-define(FATE_MAP_T,     {map,     #{ A::fate_none_map_type() => B::fate_type() }}).
```

Comment:
Won't these type variables (`A` and `B`) produce weird effects if you use multiple `?FATE_MAP_T` in a type spec?

Reply:
Maybe. They should not be needed so I removed them,